ui/app/src/services/marketplace.ts

```diff
@@ -30,6 +36,33 @@ export function fetchBlueprints(options?: { type?: string; limit?: number; showI
   );
 }
 
+export function fetchMarketplacePlugins(
+  type: string,
+  keywords?: string,
+  options?: {
+    limit?: number;
+    showIncompatible?: boolean;
+  }
+): Observable<PagedArray<MarketplacePlugin>> {
+  const qs = toQueryString({ type, keywords, ...options });
+  return get(`/studio/api/2/marketplace/search${qs}`).pipe(
+    pluck('response'),
+    map(({ plugins, offset, total, limit }) => Object.assign(plugins, { total, offset, limit }))
+  );
+}
+
+export function installMarketplacePlugin(
+  siteId: string,
+  pluginId: string,
+  pluginVersion: Version
+): Observable<ApiResponse> {
+  return postJSON('/studio/api/2/marketplace/install', { siteId, pluginId, pluginVersion }).pipe(pluck('response'));
+}
```

Comment:
Don't return ApiReponse (practically never). Pluck the "meat", if there's not meat, you may `mapTo(true)`